---
title: "AssemblyComparison"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      message = FALSE, warning = FALSE)
```

## Set Up

```{r loadpackages}
library(dplyr)
library(tibble)
library(magrittr)
library(pander)
library(scales)
library(readr)
library(ggplot2)
library(reshape2)
```

## Read Representation

```{r, echo=FALSE}
readRep <- read_tsv("~/TigerSnakeTranscriptomeVG/2_assemblyComparison/readRepAllAssemblies.tsv") 
```

```{r, echo=FALSE} 
pander(readRep, justify = "lrrr",
       caption = "Percentage of read alignment back to the transcriptome produced by each method")
```

```{r, echo=FALSE, fig.cap="*Bar graph displaying the percentage of read alignment back to each transcriptome assembly*"}
readRep %>%
  melt(id.vars = "Sample", value.name = "Percent", variable.name = "Method") %>%
  ggplot(aes(x = Sample, y = Percent, fill = Method)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_fill_manual(values = c(rgb(0.8, 0, 0), rgb(0.2, 0.6, 0.2), rgb(0.2, 0.1, 0.7))) +
  labs(y = "Percent Represented",
       fill = "Assembly Method") +
  theme_bw()
```

## Basic Comparative Assembly Statistics

```{r, echo=FALSE}
contigStats <- data.frame(
  "SoapDenovoTrans" = c("Total Genes" = 660587,
                          "Total Transcripts" = 660587,
                          "GC Percent" = 42.02,
                          "N50" = 346,
                          "Median Contig Length" = 106,
                          "Mean Contig Length" = 213.12,
                          "Total Assembled Bases" = 140782291),
  "TrinityDenovo" = c("Total Genes" = 177582,
                 "Total Transcripts" = 293817,
                 "GC Percent" = 43.27, 
                 "N50" = 2057,
                 "Median Contig Length" = 399,
                 "Mean Contig Length" = 940.29,
                 "Total Assembled Bases" = 276273069),
  "TrinityDenovoCDHIT" = c("Total Genes" = 176747,
                            "Total Transcripts" = 221351,
                            "GC Percent" = 42.89,
                            "N50" = 1297,
                            "Median Contig Length" = 353,
                            "Mean Contig Length" = 722.21,
                            "Total Assembled Bases" = 159861808)) %>%
  rownames_to_column("Measure") %>%
  as_data_frame()
```

```{r, echo=FALSE}
contigStats %>%
  pander(caption = "Comparative assembly statistical measures to assess quality of transcriptomes produced by various transcriptome assembly tools",
         justify = "lrrr",
         big.mark = ",")
```

## Coverage of the Transcriptome

```{r, echo=FALSE}
covTrans <- data.frame(
  "SoapDenovoTrans" = c(`No. bases assembled (mil)` = 140,
                           `No. read bases (mil)` = 67754,
                           Coverage = 480),
  "TrinityDenovo" = c(`No. bases assembled (mil)` = 276,
                        `No. read bases (mil)` = 67754,
                        Coverage = 245),
  "TrinityDenovoCDHIT" = c(`No. bases assembled (mil)` = 160,
                                `No. read bases (mil)` = 67754,
                                 Coverage = 423)) %>%
  as.data.frame()
```

```{r, echo=FALSE}
  pander(covTrans,
         split.table = Inf,
         justify = "lrrr",
         big.mark = ",",
         caption = "Coverage of the assembled transcriptome")
```

